TicTacToe.c

```diff
@@ -0,0 +1,128 @@
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// Ayisha Sameera,GET.
+// ------------------------------------------------------------------
+// TicTacToe.c
+// Program on branch Test3.1
+// Program implements TiTacToe with 2 or 1 player mode.
+// Program displays the board upon valid moves.
+// ------------------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <io.h>
+#include <fcntl.h>
+#include <string.h>
+
+#define DIA1 6
+#define DIA2 7
+
+static int TicTacToe (wchar_t* grid, int index, bool isPlayer1) {
+   // 3 x 3 Grid consists of three rows, three columns and two diagonals(8 possibilities)
+   static int Count[8][2] = {
+      {0,0},  // Row 1 { Count of Player 1 symbols in row1 ,Count of Player 2 symbols in row1 }
+      {0,0},  // Row 2                  .
+      {0,0},  // Row 3                  .
+      {0,0},  // Column 1               .
+      {0,0},  // Column 2               .
+      {0,0},  // Column 3               .
+      {0,0},  // Diagonal 1 (Left to right, Box positions 1-5-9)
+      {0,0}   // Disgonal 2 (Right to left, Box positions 3-5-7)
+   };
+   int rowNo = index / 3, colNo = index % 3 + 3;
+   int player = isPlayer1 ? 0 : 1;
+   Count[rowNo][player] += 1;
+   Count[colNo][player] += 1;
+   if (index % 2 == 0) {
+      switch (index) {
+         case 0: Count[DIA1][player] += 1; break;  // Diagonal 1, index positions 0-4-8
+         case 2: Count[DIA2][player] += 1; break;  // Diagonal 2, index positions 2-4-6
+         case 4: Count[DIA1][player] += 1; Count[DIA2][player] += 1; break;
+         case 6: Count[DIA2][player] += 1; break;
+         case 8: Count[DIA1][player] += 1; break;
+      }
+   }
+   if (Count[rowNo][player] >= 3 || Count[colNo][player] >= 3 ||
+       Count[DIA1][player] >= 3 || Count[DIA2][player] >= 3) return isPlayer1 ? 1 : 2;
+   return 0;
+}
+
+// Returns first non-space input character obtained from the user.
+static char Get_Input (wchar_t* prompt) {
+   char inChar[20];
+   do {
+      wprintf (L"%s ", prompt);
+      fgets (inChar, sizeof (inChar), stdin);
+      // If the input exceeds 20, clear input buffer.
+      if (inChar[strlen (inChar) - 1] != '\n') while (getchar () != '\n');
+   } while (inChar[0] == '\n' || strlen (inChar) > 2);
+   return inChar[0];
+}
+
+// Gets two non-space unique symbols to represent player 1 and 2.
+static void Get_Symbol (char* s1, char* s2) {
+   *s1 = Get_Input (L"Enter symbol 1:");
+   *s2 = Get_Input (L"Enter symbol 2:");
+   while (*s1 == *s2) *s2 = Get_Input (L"Please enter a unique symbol. Enter symbol 2:");
+}
+
+// Prints 3 x 3 game board.
+static void PrintBoard (wchar_t* grid) {
+   wprintf (L"\u250F\u2501\u2501\u2501\u2533\u2501\u2501\u2501\u2533\u2501\u2501\u2501\u2513\n\u2503");

```

Comment:
You can directly copy paste symbols. It is more readable than corresponding unicode.